vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go

```diff
@@ -0,0 +1,322 @@
+	found, err := scaleUpStatefulSet(kubeClient, labelSelector, namespace, log)
+	if err != nil {
+		return err
+	} else if !found {
+		found, err = scaleUpDeployment(kubeClient, labelSelector, namespace, log)
+		if err != nil {
+			return err
+		} else if !found {
+			return errors.Errorf("couldn't find a paused vcluster %s in namespace %s. Make sure the vcluster exists and was paused previously", name, namespace)
+		}
+
+		// scale down kube api server
+		_, err = scaleUpDeployment(kubeClient, "app=vcluster-api,release="+name, namespace, log)
+		if err != nil {
+			return err
+		}
+
+		// scale down kube controller
+		_, err = scaleUpDeployment(kubeClient, "app=vcluster-controller,release="+name, namespace, log)
+		if err != nil {
+			return err
+		}
+
+		// scale down etcd
+		_, err = scaleUpStatefulSet(kubeClient, "app=vcluster-etcd,release="+name, namespace, log)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func scaleUpDeployment(kubeClient kubernetes.Interface, labelSelector string, namespace string, log log.Logger) (bool, error) {
+	list, err := kubeClient.AppsV1().Deployments(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return false, err
+	} else if len(list.Items) == 0 {
+		return false, nil
+	}
+
+	for _, item := range list.Items {
+		if item.Annotations == nil || item.Annotations[constants.PausedAnnotation] != "true" {
+			return false, nil
+		}
+
+		originalObject := item.DeepCopy()
+
+		replicas := 1
+		if item.Annotations[constants.PausedReplicasAnnotation] != "" {
+			replicas, err = strconv.Atoi(item.Annotations[constants.PausedReplicasAnnotation])
+			if err != nil {
+				log.Warnf("error parsing old replicas: %v", err)
+				replicas = 1
+			}
+		}
+
+		replicas32 := int32(replicas)
+		delete(item.Annotations, constants.PausedAnnotation)
+		delete(item.Annotations, constants.PausedReplicasAnnotation)
+		item.Spec.Replicas = &replicas32
+
+		patch := client.MergeFrom(originalObject)
+		data, err := patch.Data(&item)
+		if err != nil {
+			return false, errors.Wrap(err, "create deployment patch")
+		}
+
+		// patch deployment
+		_, err = kubeClient.AppsV1().Deployments(namespace).Patch(context.TODO(), item.Name, patch.Type(), data, metav1.PatchOptions{})
+		if err != nil {
+			return false, errors.Wrap(err, "patch deployment")
+		}
+	}
+
+	return true, nil
+}
+
+func scaleUpStatefulSet(kubeClient kubernetes.Interface, labelSelector string, namespace string, log log.Logger) (bool, error) {
+	list, err := kubeClient.AppsV1().StatefulSets(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return false, err
+	} else if len(list.Items) == 0 {
+		return false, nil
+	}
+
+	for _, item := range list.Items {
+		if item.Annotations == nil || item.Annotations[constants.PausedAnnotation] != "true" {
+			return false, nil
+		}
+
+		originalObject := item.DeepCopy()
+
+		replicas := 1
```

Comment:
If the replica was a different number before pause, shouldn't we bring up the same count?

Reply:
Yes! If you look a few lines down you'll see that the old replica count is retrieved from an annotation. 1 is just the default.

Reply:
sorry, missed to see it previously.